src/filters/factory.rs

```diff
@@ -63,45 +63,61 @@ pub trait FilterFactory: Sync + Send {
 
     /// Returns the [`ConfigType`] from the provided Option, otherwise it returns
     /// Error::MissingConfig if the Option is None.
-    fn require_config<'a, 'b>(
-        &'a self,
-        config: Option<ConfigType<'b>>,
-    ) -> Result<ConfigType<'b>, Error> {
+    fn require_config(&self, config: Option<ConfigType>) -> Result<ConfigType, Error> {
         config.ok_or_else(|| Error::MissingConfig(self.name()))
     }
 }
 
 /// Arguments needed to create a new filter.
-pub struct CreateFilterArgs<'a> {
+pub struct CreateFilterArgs {
```

Comment:
I've added the label, which I believe is what @markmandel is using.

Reply:
Yeah we need the label too! What I meant was in this case the label isn't enough since the breakage isn't obvious at all from the PR - usually with a release folks will go through the breaking PRs (which is where the label comes in) and it should be clear what broke. Here that won't be possible at all without reading the code which would be harsh, so let's mention that this PR does in fact contain a breaking change since adding a Matches filter isn't breaking in itself.

Reply:
Yes please - add a note about this in the description, so when i write the release notes I won't forget 👍🏻